test/entity.ts

```diff
@@ -66,6 +67,102 @@ describe('entity', () => {
       const int = new entity.Int(value);
       assert.strictEqual(int.value, value.toString());
     });
+
+    it('should store the stringified value from valueProto object', () => {
+      const valueProto = {
+        valueType: 'integerValue',
+        integerValue: 8,
+      };
+      const int = new entity.Int(valueProto);
+      assert.strictEqual(int.value, valueProto.integerValue.toString());
+    });
+
+    describe('valueOf', () => {
+      let valueProto: {};
+      beforeEach(() => {
+        valueProto = {
+          valueType: 'integerValue',
+          integerValue: 8,
+        };
+      });
+
+      it('should throw if integerTypeCastFunction is not provided', () => {
+        assert.throws(
+          () => new entity.Int(valueProto, {}).valueOf(),
```

Comment:
should we default to providing a casting function that throws if the integer is larger than max int?

Reply:
(seems like `valueOf` should work by default).

Reply:
+1. This seems good. It makes the default the same behavior of the original except we throw on corruption of the data.

Reply:
Looking closer it seems this is just validating if an empty dict is given as a param it fails as that isn't valid.